posthog/storage/session_recording_v2_object_storage.py

```diff
@@ -199,6 +270,176 @@ def is_lts_enabled(self) -> bool:
         return bool(settings.SESSION_RECORDING_V2_S3_LTS_PREFIX)
 
 
+class AsyncSessionRecordingV2ObjectStorage:
+    def __init__(self, aws_client, bucket: str) -> None:
+        self.aws_client = aws_client
+        self.bucket = bucket
+
+    async def read_bytes(self, key: str, first_byte: int, last_byte: int) -> bytes | None:
+        s3_response = {}
+        try:
+            kwargs = {
+                "Bucket": self.bucket,
+                "Key": key,
+                "Range": f"bytes={first_byte}-{last_byte}",
+            }
+            s3_response = await self.aws_client.get_object(**kwargs)
+            return await s3_response["Body"].read()
+        except Exception as e:
+            logger.exception(
+                "session_recording_v2_object_storage.read_failed",
+                bucket=self.bucket,
+                file_name=key,
+                error=e,
+                s3_response=s3_response,
+            )
+            return None
+
+    async def read_all_bytes(self, key: str) -> bytes | None:
+        s3_response = {}
+        try:
+            kwargs = {
+                "Bucket": self.bucket,
+                "Key": key,
+            }
+            s3_response = await self.aws_client.get_object(**kwargs)
+            return await s3_response["Body"].read()
+        except Exception as e:
+            logger.exception(
+                "session_recording_v2_object_storage.read_failed",
+                bucket=self.bucket,
+                file_name=key,
+                error=e,
+                s3_response=s3_response,
+            )
+            return None
+
+    async def write(self, key: str, data: bytes) -> None:
+        s3_response = {}
+        try:
+            s3_response = await self.aws_client.put_object(
+                Bucket=self.bucket,
+                Key=key,
+                Body=data,
+            )
+        except Exception as e:
+            logger.exception(
+                "session_recording_v2_object_storage.write_failed",
```

Comment:
```suggestion
                "session_recording_v2_async_object_storage.write_failed",
```

same throughout to help the 2am stressed incident person :)